runtime/modules/xml/core/src/main/java/io/atlasmap/xml/inspect/SchemaInspector.java

```diff
@@ -317,44 +321,48 @@ private void printSchemaSet(XSSchemaSet schemaSet) throws Exception {
         }
     }
 
-    private void printComplexType(XSComplexType complexType, String rootName, XmlComplexType xmlComplexType)
+    private void printComplexType(XSComplexType complexType, String rootName, XmlComplexType xmlComplexType, Set<String> cachedComplexType)
             throws Exception {
         printAttributes(complexType, rootName, xmlComplexType);
         XSParticle particle = complexType.getContentType().asParticle();
         if (particle != null) {
-            printParticle(particle, rootName, xmlComplexType);
+            printParticle(particle, rootName, xmlComplexType, cachedComplexType);
         }
     }
 
-    private void printParticle(XSParticle particle, String rootName, XmlComplexType xmlComplexType)
+    private void printParticle(XSParticle particle, String rootName, XmlComplexType xmlComplexType, Set<String> cachedComplexType)
             throws Exception {
         XSTerm term = particle.getTerm();
         if (term.isModelGroup()) {
             XSModelGroup group = term.asModelGroup();
-            printGroup(group, rootName, xmlComplexType);
+            printGroup(group, rootName, xmlComplexType, cachedComplexType);
         } else if (term.isModelGroupDecl()) {
-            printGroupDecl(term.asModelGroupDecl(), rootName, xmlComplexType);
+            printGroupDecl(term.asModelGroupDecl(), rootName, xmlComplexType, cachedComplexType);
         } else if (term.isElementDecl()) {
             CollectionType collectionType = getCollectionType(particle);
-            printElement(term.asElementDecl(), rootName, xmlComplexType, collectionType);
+            printElement(term.asElementDecl(), rootName, xmlComplexType, collectionType, cachedComplexType);
         }
     }
 
-    private void printGroup(XSModelGroup modelGroup, String rootName, XmlComplexType xmlComplexType)
+    private void printGroup(XSModelGroup modelGroup, String rootName, XmlComplexType xmlComplexType, Set<String> cachedComplexType)
             throws Exception {
         // this is the parent of the group
         for (XSParticle particle : modelGroup.getChildren()) {
-            printParticle(particle, rootName, xmlComplexType);
+            Set<String> cachedTypeCopy = new HashSet<>(cachedComplexType);
```

Comment:
It's not obvious, why you need to make a copy of cachedComplexType here.

Reply:
Ah I should have left a comment - this is to avoid caching siblings, as we don't support expanding cached field in UI yet (#255). This is another reason we should address #579.

Reply:
Thanks! Makes sense. Indeed a comment would be great.